cmd/otelcol/main.go

```diff
@@ -169,66 +155,119 @@ func checkRuntimeParams() {
 	// Check if memory ballast flag was passed
 	// If so, ensure memory ballast env var is not set
 	// Then set memory ballast and limit properly
-	ballastSize := getKeyValue(args, "--mem-ballast-size-mib")
+	ballastSize := getKeyValue(os.Args[1:], "--mem-ballast-size-mib")
 	if ballastSize != "" {
-		config = os.Getenv(ballastEnvVarName)
-		if config != "" {
-			log.Fatalf("Both %v and '--config' were specified, but only one is allowed", ballastEnvVarName)
+		if os.Getenv(ballastEnvVarName) != "" {
+			log.Fatalf("Both %v and '--mem-ballast-size-mib' were specified, but only one is allowed", ballastEnvVarName)
 		}
 		os.Setenv(ballastEnvVarName, ballastSize)
 	}
 	setMemoryBallast(memTotalSizeMiB)
 	setMemoryLimit(memTotalSizeMiB)
 }
 
-// Validate and set the configuration
-func setConfig() {
-	// Check if the config is specified via the env var.
-	config := os.Getenv(configEnvVarName)
-	// If not attempt to use a default config; supports Docker and local
-	if config == "" {
-		_, err := os.Stat(defaultDockerSAPMConfig)
-		if err == nil {
-			config = defaultDockerSAPMConfig
+// Validate then set config file path environment variable to specified config file path flag.
```

Comment:
I don't think this is "necessary" anymore since the function that set the flag from the env var value was removed in these changes.

Reply:
My understanding is that we always sync flag --config with env var SPLUNK_CONFIG with --config having the highest priority. So, here start by saying SPLUNK_CONFIG is set to "specified" --config value period. And in the next sentence I say --config is set to "specified" SPLUNK_CONFIG value if empty. Then defaults if --config, SPLUNK_CONFIG and SPLUNK_CONFIG_YAML are not specified..